services/src/main/scala/cromwell/pipeline/service/ProjectService.scala

```diff
@@ -40,22 +44,34 @@ object ProjectService {
     new ProjectService {
 
       def getUserProjectById(projectId: ProjectId, userId: UserId): Future[Project] =
-        projectRepository.getProjectById(projectId).flatMap(project => getForUserOrFail(project.toSeq, userId))
+        projectRepository
+          .getProjectById(projectId)
+          .recoverWith {
+            case _ => Future.failed(InternalError("Failed to find user project due to unexpected internal error"))
```

Comment:
Maybe we can make a helper methods to reduce duplication
```scala
private def internalError(action: String): Future[Nothing] =
  Future.failed(InternalError(s"Failed to $action due to unexpected internal error"))
```
```suggestion
            case _ => internalError("find user project")
```

Reply:
I`ll do this ASAP.

Reply:
Done.